Added macro support, compatibility

wbamberg opened the pull request:
1. Added extension_types support
2. Added compat table generation
3. Changed Chromium acknowledgements
4. Added calls to xref macro "wexref"
5. Added tags to JSON metadata
6. Added calls to insert sidebar using WebExtensionSidebar macro

Commit:
1. Added extension_types support
2. Added compat table generation
3. Changed Chromium acknowledgements
4. Added calls to xref macro "wexref"
5. Added tags to JSON metadata
6. Added calls to insert sidebar using WebExtensionSidebar macro

bill-mccloskey commented:
The columns in the compat table don't seem like the right ones to me. Can we just include Firefox, Chrome, and Edge? Those are the only ones likely to support this API in the future.

Also, I think we're supposed to have more tags. https://developer.mozilla.org/en-US/docs/MDN/Contribute/Howto/Write_an_API_reference#Tagging

wbamberg commented:
> The columns in the compat table don't seem like the right ones to me.
> Can we just include Firefox, Chrome, and Edge? Those are the only
> ones likely to support this API in the future.

I just copied the table from https://developer.mozilla.org/en-US/docs/MDN/Contribute/Structures/Compatibility_tables, but I agree, there's no point in including browsers that will never support this.

Which mobile browsers do we care about?

> Also, I think we're supposed to have more tags.
> https://developer.mozilla.org/en-US/docs/MDN/Contribute/Howto/Write_an_API_reference#Tagging

OK, I'll add those.

Commit:
1) added more tags
2) changed browsers included in compat table
3) rewrite $(ref-style references to use wexref
4) rename WebExtensionSidebar macro to AddonSidebar

Commit:
convert  into xref links; fix event listener pages
